### DISCUSSION TITLE
Blocks filter supposed to give whole Para for conversion into Div, but ends up putting each word into its own Div

### DISCUSSION THREAD

**Discussion by ElDuderino321** [jgm/pandoc#10310] [General]
I had a question a few weeks ago about converting paragraphs into a custom-div style for .docx convertion (s. below), and somebody gave me some code. I only tried it out a few days ago, and realized the result is not as expected. Instead of converting a whole Para block into the div, each word is div'ed individually (making the resulting .docx unsable). If I look at the AST, each word ends up as its own Plain element, which in turn is in a div.

I am grateful for the helpful suggestion, but I lack the Lua skills (so far) to figure out what went wrong. (As far as I can tell, the code is correct!)

Here is the code I used, based on the suggestion from another user (s. below):

```lua
function Blocks (blocks)
   for i, block in ipairs(blocks) do
     if i >= 2 and block.t == "Para" and blocks[i-1].t == "Heading" then
         blocks[i] = pandoc.Div(block.content, {["custom-style"] = "Paragraph"})
     else
         blocks[i] = pandoc.Div(block.content, {["custom-style"] = "New Paragraph"})
     end
   end
   return blocks
end
```

Here is the original post and issue:

> Use the `Blocks` filter and iterate over the top level blocks yourself instead of letting Pandoc throw `Para` blocks to your filter willy-nilly. Change them to what you want them to be in place, something like this:
> 
> ```lua
> local function Blocks (blocks)
>    for i, block in ipairs(blocks) do
>       if block.t == "Para" then
>          blocks[i] = pandoc.Div(block.content, {...})
>       end
>    end
>    return blocks
> end
> ```

I actually only now got around to trying this out, and in the output, every word is now its own paragraph (albeit in the correct style). Looking at the resulting AST, each div contains Plain elements, which contains the Str and Space elements. Is that related to the problem?

(I also had to remove the "local" for the filter to function. Otherwise, it did not nothing. Is that a pandoc thing?)

_Originally posted by @ElDuderino321 in https://github.com/jgm/pandoc/discussions/10124#discussioncomment-10908972_

**Comment by jgm** [comment]
```lua
pandoc.Div(block.content
```
When the block is of type `Para`, `content` will be a list of inline elements (e.g. one for each word), not a list of block elements.  pandoc.Div implicitly converets these to blocks.  You probably want something like
```lua
pandoc.Div(pandoc.Plain(block.content)
```

**Reply by ElDuderino321** [reply]
Thank you! That means I would need to iterate over the Inline elements (block.content) and gather them inside a Plain element?
